driver/controller.go

```diff
@@ -416,34 +439,81 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 // calculateStorageGB extracts the storage size in GB from the given capacity
 // range. If the capacity range is not satisfied it returns the default volume
 // size.
-func calculateStorageGB(capRange *csi.CapacityRange) (int, error) {
+func calculateStorageGB(capRange *csi.CapacityRange, storageType string) (int, error) {
+	sizeIncrements := SSDStepSizeGB
+	if storageType == "bulk" {
+		sizeIncrements = BulkStepSizeGB
+	}
 	if capRange == nil {
-		return SSDStepSizeGB, nil
+		return sizeIncrements, nil
+	}
+
+	// Volume MUST be at least this big. This field is OPTIONAL.
+	// A value of 0 is equal to an unspecified field value.
+	// The value of this field MUST NOT be negative.
+	requiredBytes := capRange.GetRequiredBytes()
+	requiredSet := 0 < requiredBytes
+
+	// Volume MUST not be bigger than this. This field is OPTIONAL.
+	// A value of 0 is equal to an unspecified field value.
+	// The value of this field MUST NOT be negative.
+	limitBytes := capRange.GetLimitBytes()
+	limitSet := 0 < limitBytes
+
+	if !requiredSet && !limitSet {
+		return sizeIncrements, nil
+	}
+	if requiredSet && limitSet && limitBytes < requiredBytes {
+		return 0, fmt.Errorf("limit (%v) can not be less than required (%v) size", formatBytes(limitBytes), formatBytes(requiredBytes))
 	}
 
-	minBytes := capRange.RequiredBytes
-	// limitBytes might be zero
-	maxBytes := capRange.LimitBytes
-	if minBytes == 0 {
-		minBytes = SSDStepSizeGB * GB
+	if requiredSet && !limitSet && requiredBytes < (int64(sizeIncrements)*GB) {
+		return 0, fmt.Errorf("required (%v) can not be less than minimum supported volume size for type '%s' (%v)", formatBytes(requiredBytes), storageType, formatBytes(int64(sizeIncrements)*GB))
```

Comment:
I don't think this is correct. Shouldn't it be possible to create a volume with required bytes == 500mb and the API should create a volume of 1 GB?

Reply:
> I don't think this is correct. Shouldn't it be possible to create a volume with required bytes == 500mb and the API should create a volume of 1 GB?

I think this would make sense, yes. I'll update the logic